tests/commands/test_build.py

```diff
@@ -222,15 +234,11 @@ def mock_is_base_available():
 
 def test_build_error_without_metadata_yaml(basic_project):
     """Validate error if trying to build project without metadata.yaml."""
-    metadata = basic_project / METADATA_FILENAME
+    metadata = basic_project / "metadata.yaml"
     metadata.unlink()
 
-    config = load(basic_project)
-
-    msg = re.escape(
-        "Cannot read the metadata.yaml file: FileNotFoundError(2, 'No such file or directory')"
-    )
-    with pytest.raises(CraftError, match=msg):
+    with pytest.raises(CraftError):
+        config = load(basic_project)
         get_builder(config)
```

Comment:
Please ensure we still check the error message.

```suggestion
    with pytest.raises(CraftError) as exc_info:
        config = load(basic_project)
        get_builder(config)
        
    assert exc_info.value.args[0] == (
        "Cannot read the metadata.yaml file: FileNotFoundError(2, 'No such file or directory')"
    )
```